ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -0,0 +1,255 @@
+
+    property string inputTokenKey
+    onInputTokenKeyChanged: Qt.callLater(d.setSelectedHoldingId, inputTokenKey, Constants.TokenType.ERC20)
+    property string inputTokenAmount
+    onInputTokenAmountChanged: Qt.callLater(() => amountToSendInput.input.text =
+                                            Number(inputTokenAmount).toLocaleString(Qt.locale(), 'f', -128))
+
+    property int swapSide: SwapInputPanel.SwapSide.Pay
+
+    // output API
+    readonly property string selectedHoldingId: d.selectedHoldingId
+    readonly property double cryptoValue: amountToSendInput.cryptoValueToSendFloat
+    readonly property string cryptoValueRaw: amountToSendInput.cryptoValueToSend
+    readonly property bool cryptoValueValid: amountToSendInput.inputNumberValid
+
+    // visual properties
+    property int swapExchangeButtonWidth: 44
+    property string caption: swapSide === SwapInputPanel.SwapSide.Pay ? qsTr("Pay") : qsTr("Receive")
+
+    enum SwapSide {
+        Pay = 0,
+        Receive = 1
+    }
+
+    padding: Style.current.padding
+
+    // by design
+    implicitWidth: 492
+    implicitHeight: 131
+
+    QtObject {
+        id: d
+
+        function setSelectedHoldingId(holdingId, holdingType) {
+            let holding = root.store.getHolding(holdingId, holdingType)
+            d.selectedHoldingId = holdingId
+            d.setSelectedHolding(holding, holdingType)
+        }
+
+        function setSelectedHolding(holding, holdingType) {
+            d.selectedHoldingType = holdingType
+            d.selectedHolding = holding
+            let selectorHolding = root.store.holdingToSelectorHolding(holding, holdingType)
+            holdingSelector.setSelectedItem(selectorHolding, holdingType)
+        }
+
+        property var selectedHolding: null
+        property var selectedHoldingType: Constants.TokenType.Unknown
+        property string selectedHoldingId
+
+        readonly property bool isSelectedHoldingValidAsset: !!selectedHolding && selectedHoldingType === Constants.TokenType.ERC20
+        readonly property double maxFiatBalance: isSelectedHoldingValidAsset ? selectedHolding.currentCurrencyBalance : 0
+        readonly property double maxCryptoBalance: isSelectedHoldingValidAsset ? selectedHolding.currentBalance : 0
+        readonly property double maxInputBalance: amountToSendInput.inputIsFiat ? maxFiatBalance : maxCryptoBalance
+        readonly property string inputSymbol: amountToSendInput.inputIsFiat ? root.store.currencyStore.currentCurrency :
+                                                                              !!d.selectedHolding && !!d.selectedHolding.symbol ? d.selectedHolding.symbol: ""
+
+        function prepareForMaxSend(value, symbol) {
+            if (symbol !== Constants.ethToken) {
+                return value
+            }
+
+            return value - Math.max(0.0001, Math.min(0.01, value * 0.1))
+        }
+    }
+
+    background: Shape {
```

Comment:
can we move background to a different file? will help this file be more clean.

Reply:
Not so sure... the background is very specific to this component and wouldn't be useful elsewhere

Reply:
true, was just wondering it would make this file a little less small but I a, okay with whatever :)